src/auth/guards/roles.guard.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable, CanActivate, ExecutionContext } from '@nestjs/common';
+import { Reflector } from '@nestjs/core';
+import { UserRoleEnum } from '../../users/enum/user-role.enum';
+import { ROLES_KEY } from '../roles.decorator';
+
+@Injectable()
+export class RolesGuard implements CanActivate {
+  constructor(private reflector: Reflector) {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const requiredRoles = this.reflector.getAllAndOverride<UserRoleEnum[]>(
+      ROLES_KEY,
+      [context.getHandler(), context.getClass()],
+    );
+    if (!requiredRoles) {
+      return true;
+    }
+    const { user } = context.switchToHttp().getRequest();
+    return requiredRoles.some((role) => user.roles?.includes(role));
```

Comment:
Let's change like this.
```ts
const { user } = context.switchToHttp().getRequest();
const { role } = await this.authService.findOne(user.id);
return requiredRoles.some((requiredRole) => requiredRole === role);
```

Also, please adjust async function and implement AuthService.findOne()